I think that's everything except for @nalimilan's suggestion

> BTW, we should support/recommend passing a tuple rather than a vector of functions, since that would allow inference to reason about the return type, and (I think) use a concrete element type rather than Any.

Here's what I tried
```julia
function colwise{T}(fns::T, d::AbstractDataTable)
    @assert all(f -> isa(f, Function), fns)
    x = [f(d[i]) for f in fns, i in 1:ncol(d)]
    if eltype(x) <: Nullable
        return NullableArray(x)
    else
        return x
    end
end
```

which behaved like so

```julia
julia> using DataTables
INFO: Precompiling module DataTables.

julia> colwise((sum, length), DataTable(A = 1:10, B = 11:20))
2×2 Array{Any,2}:
   55    155
 10    10

julia> colwise(("Bob", :Susie), DataTable(A = 1:10, B = 11:20))
ERROR: AssertionError: all((f->begin  # /Users/Cameron/.julia/v0.5/DataTables/src/groupeddatatable/grouping.jl, line 260:
            isa(f,Function)
        end),fns)
 in colwise(::Tuple{String,Symbol}, ::DataTables.DataTable) at /Users/Cameron/.julia/v0.5/DataTables/src/groupeddatatable/grouping.jl:260

julia> [typeof(x) for x in colwise((sum, length), DataTable(A = 1:10, B = 11:20))]
2×2 Array{DataType,2}:
 Nullable{Int64}  Nullable{Int64}
 Int64            Int64
```

--- a/src/groupeddatatable/grouping.jl
+++ b/src/groupeddatatable/grouping.jl
@@ -257,12 +257,15 @@ function colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable)
     end
 end
 function colwise(fns::Tuple, d::AbstractDataTable)
-    @assert all(f -> isa(f, Function), fns) || ArgumentError("Not all tuple elements are functions")
-    x = [f(d[i]) for f in fns, i in 1:ncol(d)]
-    if eltype(x) <: Nullable
-        return NullableArray(x)
+    if any(f -> !isa(f, Function), fns)
+        throw(ArgumentError("Not all tuple elements are functions"))
     else
-        return x
+        x = [f(d[i]) for f in fns, i in 1:ncol(d)]
+        if eltype(x) <: Nullable
+            return NullableArray(x)
+        else
+            return x
+        end
     end
 end
 colwise{T<:Function}(fns::Vector{T}, gd::GroupedDataTable) = map(colwise(fns), gd)

--- a/src/groupeddatatable/grouping.jl
+++ b/src/groupeddatatable/grouping.jl
@@ -256,16 +256,12 @@ function colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable)
         return x
     end
 end
-function colwise(fns::Tuple, d::AbstractDataTable)
-    if any(f -> !isa(f, Function), fns)
-        throw(ArgumentError("Not all tuple elements are functions"))
+function colwise{N}(fns::NTuple{N,Function}, d::AbstractDataTable)
+    x = [f(d[i]) for f in fns, i in 1:ncol(d)]
+    if eltype(x) <: Nullable
+        return NullableArray(x)
     else
-        x = [f(d[i]) for f in fns, i in 1:ncol(d)]
-        if eltype(x) <: Nullable
-            return NullableArray(x)
-        else
-            return x
-        end
+        return x
     end
 end
 colwise{T<:Function}(fns::Vector{T}, gd::GroupedDataTable) = map(colwise(fns), gd)

--- a/src/groupeddatatable/grouping.jl
+++ b/src/groupeddatatable/grouping.jl
@@ -256,6 +256,15 @@ function colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable)
         return x
     end
 end
+function colwise(fns::Tuple, d::AbstractDataTable)
+    @assert all(f -> isa(f, Function), fns) || ArgumentError("Not all tuple elements are functions")
+    x = [f(d[i]) for f in fns, i in 1:ncol(d)]
+    if eltype(x) <: Nullable
+        return NullableArray(x)
+    else
+        return x
+    end
+end
 colwise{T<:Function}(fns::Vector{T}, gd::GroupedDataTable) = map(colwise(fns), gd)
 colwise{T<:Function}(fns::Vector{T}) = x -> colwise(fns, x)
 

--- a/test/grouping.jl
+++ b/test/grouping.jl
@@ -38,7 +38,7 @@ module TestGrouping
     answer = Any[Nullable(20) Nullable(12) Nullable(-0.4283098098931877);
                  8            8            8                            ]
     @test isequal(cw, answer)
-    @test_throws ArgumentError colwise(("Bob", :Susie), DataTable(A = 1:10, B = 11:20))
+    @test_throws MethodError colwise(("Bob", :Susie), DataTable(A = 1:10, B = 11:20))
     # colwise(::Function, ::AbstractDataTable)
     cw = colwise(sum, dt)
     @test all(T -> isa(T, Nullable), cw)

--- a/test/grouping.jl
+++ b/test/grouping.jl
@@ -38,6 +38,7 @@ module TestGrouping
     answer = Any[Nullable(20) Nullable(12) Nullable(-0.4283098098931877);
                  8            8            8                            ]
     @test isequal(cw, answer)
+    @test_throws ArgumentError colwise(("Bob", :Susie), DataTable(A = 1:10, B = 11:20))
     # colwise(::Function, ::AbstractDataTable)
     cw = colwise(sum, dt)
     @test all(T -> isa(T, Nullable), cw)

--- a/test/grouping.jl
+++ b/test/grouping.jl
@@ -25,14 +25,24 @@ module TestGrouping
 
     byf = by(dt, :a, dt -> DataTable(bsum = sum(dt[:b])))
 
+    # colwise(::Vector{<:Function}, ::AbstractDataTable)
     cw = colwise([sum], dt)
     @test isa(cw, NullableArray{Any, 2})
     @test size(cw) == (1,ncol(dt))
     answer = NullableArray([20 12 -0.4283098098931877])
     @test isequal(cw, answer)
+    # colwise(::Tuple{<:Function}, ::AbstractDataTable)
+    cw = colwise((sum, length), dt)
+    @test isa(cw, Array{Any, 2})
+    @test size(cw) == (2,ncol(dt))
+    answer = Any[Nullable(20) Nullable(12) Nullable(-0.4283098098931877);
+                 8            8            8                            ]
+    @test isequal(cw, answer)
+    # colwise(::Function, ::AbstractDataTable)
     cw = colwise(sum, dt)
     @test all(T -> isa(T, Nullable), cw)
-    @test isequal(cw, squeeze(answer, 1))
+    answer = NullableArray([20, 12, -0.4283098098931877])
+    @test isequal(cw, answer)
 
     # groupby() without groups sorting
     gd = groupby(dt, cols)

